pubsub/tests/unit/subscriber_test.py

```diff
@@ -270,36 +271,34 @@ async def f(*args, **kwargs):
         assert producer_task.exception()
 
     @pytest.mark.asyncio
-    async def test_producer_gracefully_shutdowns(subscriber_client):
-        mock = MagicMock()
-        def g():
-            yield 1
-            mock()
+    async def test_producer_gracefully_shutsdown(subscriber_client):
+        def f():
```

Comment:
Do we expect `f` to be called twice here? Just wondering as this first raises the CancelledError, and _then_ returns an integer. But it looks like reversing this (returning integer, _then_ raising) also results in the test passing.

Also maybe using `Mock(side_effect=(1, asyncio.CancelledError))` is clearer than adding the function attribute?

Reply:
Yeah, first we need to raise to simulate being cancelled and then it should be called normally to not raise during graceful shutdown.

Ha, I didn't know `side_effect` work like that, awesome